File: Java/src/p_3_creational_patterns/p_7_factorymethod/DocumentFactory.java
package p_3_creational_patterns.p_7_factorymethod;

public class DocumentFactory {
    public static Document createDocument( EDocumentType type) throws Exception
    {
        if ( type.equals(EDocumentType.PDF))
            return new PDF();
        else if ( type.equals(EDocumentType.WORD))
            return new Word();

        throw new RuntimeException("Not Found");
    }
}

File: Net/03_creational_patterns/03_factorymethod_pattern/DocumentFactory.cs
namespace _03_factorymethod_pattern
{
    public enum DocumentTypes
    {
        Word,Pdf
    }

    public class DocumentFactory
    {
        public static IDocument CreateDocument(DocumentTypes documentType)
        {
            switch(documentType)
            {
                case DocumentTypes.Word:
                    return new Word();
                case DocumentTypes.Pdf:
                    return new PDF();
            }

            throw new Exception("There is no type of a object");
        }
    }
}

